### DISCUSSION TITLE
Color palettes

### DISCUSSION THREAD

**Discussion by AnnaBhlr** [brainvisa/brainvisa.github.io#151] [General]
Hi, 
I would like to change the color scheme of a sulci graph in python using one of the color palettes provided by brainvisa. Unfortunately, in the online doc of the GUI I found only a few names of color palettes (eg "Blue-Pastel-Red"). Is there a full list  of available color schemes available (without installing the GUI)?
Thanks for the help,
Anna

**Comment by denisri** [comment]
Hi,
- If you are speaking about sulci graphs, their colors are not in a palette object actually: it is in the nomenclature (`sulcal_root_colors.hie`).
- The palettes list can otherwise be queried in python like this:
  ```
  import anatomist.api as ana
  a = ana.Anatomist()
  palette_names = [p.name() for p in a.palettes().palettes()]
  ```
- it is possible to load and manipulate "nomenclature objects". They have a tree structure:
  ```
  import anatomist.api as ana
  from soma import aims
  a = ana.Anatomist()
  nom_fname = aims.carto.Paths.findResourceFile('nomenclature/hierarchy/sulcal_root_colors.hie')
  nom = a.loadObject(nom_fname)
  target_leaf = ['brain', 'hemisph_left', 'S.C._left']
  tree = nom.attributed()
  for name_level in target_leaf:
      tree = [child for child in tree.children() if child['fold_name'] == name_level][0]
  print('color:', tree['color'])
  ```
(this is done using the API of Anatomist, which is actually a GUI)
Denis

**Comment by sarafv** [comment]
Hello, 
I want to import to our toolbox  epiSTIM on 3D Slicer the meshes of the morphologist labeled sulci.
It will be nice to have the color table for the sulci.
Do the fold_names in sulcal_root_colors.hie correspond to a standard anatomy ontology?
Do the child folds inherite the colors of ancestors? because the color is not defined for all the fold_name elements 

Thanks
Sara

**Comment by denisri** [comment]
> Do the fold_names in sulcal_root_colors.hie correspond to a standard anatomy ontology?

Ahem, what do you call "standard" ? There is even no "real" official standard sulci nomenclature: there are commonly used names for the major ones, but as soon as you get into details, different teams use different names... The ontology in our .hie file is derived from names used by Jean Régis at the end of the 90's. There are differences with nomenclatures used in Ono, or Petrides, or other groups...

> Do the child folds inherite the colors of ancestors? because the color is not defined for all the fold_name elements

Yes the colors assignment is done hierarchically (the `.hie` extension stands for "hierarchy", it's a tree structure).

Denis

**Comment by sarafv** [comment]
yes standard ==> official, it coudn't find the good word. I dont know the anatomy and the differents ontologies or terminologies of the sulci, but il would be nice to have it in a .ctbl file for the sulci colos and a terminology/ontologie  json file
Thanks again
Sara

**Comment by denisri** [comment]
`.ctbl` is a 3D Slicer format, right ? I guess you can build it easily from the `.hie` of BrainVisa but I don't have a tool ready for that.
